src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/PerformanceAnalyzerMetrics.java

```diff
@@ -82,7 +82,13 @@ public static String getCurrentTimeMetric() {
     public static String generatePath(long startTime, String... keysPath) {
         StringBuilder stringBuilder = new StringBuilder(sDevShmLocation);
 
-        stringBuilder.append(String.valueOf(PerformanceAnalyzerMetrics.getTimeInterval(startTime))).append(File.separator);
+        char lastChar = stringBuilder.charAt(stringBuilder.length() - 1);
```

Comment:
Instead of manually checking for file separator, can we use the Path library in Java?